tough/src/lib.rs

```diff
@@ -667,6 +667,7 @@ fn parse_url(url: Url) -> Result<Url> {
 /// Steps 0 and 1 of the client application, which load the current root metadata file based on a
 /// trusted root metadata file.
 #[expect(clippy::too_many_arguments)]
+#[allow(clippy::needless_continue)]
```

Comment:
Can you use `expect` instead of `allow`?

Reply:
Hm, tried this and apparently not. There seems to be something odd with the macos Github actions runner, it doesn't register a `needless_continue` in this function so it will fail the `expect` (and it's also had a bunch of other strange failures on this PR). I couldn't replicate this on my mac, but switching this back to `allow` to pass the workflows.

Reply:
This suggests to me that the Mac actions runner must be using an older rust toolchain.

I think it points out to me that we probably don't want this to fail if you are using an older rust toolchain, so I'm OK to leave it as-is.